components/ReportList.vue

```diff
@@ -10,32 +10,37 @@ import { useSingleReport, useCommit } from "../composables";
 import { BusinessButton, ShieldIcon } from "@axodotdev/fringe/lib";
 import { getVersionChangeText } from "../utils/versions";
 
-const { areAllEulasApproved, report } = useSingleReport();
+const { report } = useSingleReport();
 const { query } = useRoute();
 const { commit, isLoading } = useCommit();
 
-const selected = useState(() =>
+const selected = ref(
   query.name
-    ? report.value.suggestions.find(
+    ? report.value?.suggestions.find(
         (suggestion) => suggestion.name === query.name
       )
-    : report.value.suggestions[0]
+    : report.value?.suggestions[0]
 );
 
 watchEffect(async () => {
   await navigateTo({
     replace: true,
     query: {
-      name: selected.value.name,
+      name: selected.value?.name,
     },
   });
 });
 
-const isAllApproved = (dep) =>
-  areAllEulasApproved(
-    report.value.suggestions.find((suggestion) => suggestion.name === dep.name)
+const isAllApproved = (dep) => {
+  const currentReport = report.value.suggestions.find(
+    (suggestion) => suggestion.name === dep.name
   );
 
+  return currentReport.suggested_criteria.every(
+    (criteria) => report.value?.state?.[currentReport.name]?.[criteria]
```

Comment:
I was imprecise. I am asking not about the result of `every`, but the function `every` is calling: 

```
(criteria) => report.value?.state?.[currentReport.name]?.[criteria]`
```
Is the result of that lookup a Boolean or a String or an Object that we are implicitly converting to a Boolean? (I know it can be `undefined` based on the conditional lookup, and so we are at least implictly converting that.)

The reason I think this is important, even though it can seem like a fine detail, is that if there is a bug later on, knowing whether the value is being converted or is expected to be a Boolean can really help idenify places where our assumptions might be incorrect and could therefore be generating the issue. While we are neck deep in things, we may as well leave ourlseves signposts.

Reply:
yeah yeah that's super valid! Added a boolean call

I do prefer writing `Boolean(thing)` than `!!thing` because I feel like it's more straighforward what's happening, let me know your throughts there